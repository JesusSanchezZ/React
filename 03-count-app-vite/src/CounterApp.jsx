import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ({ value }) => {
    const [ counter, setCounter ] = useState(value);

    const handleAdd = ( f ) => {
        //setCounter( counter + 1 );
        setCounter( (c) => c + 1);
    };
    const handleSub = () => {
        setCounter( counter - 1);
    };
    const handleRst = () => {
        setCounter( value );
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleSub }>-1</button>
            <button aria-label='button-reset' onClick={ handleRst }>Reset</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

/**
 * Pruebas a la aplicación
 * instalación de Jest
 *  npm install --save-dev jest
 * agregamos script al package.json
 *  "test":"jest"
 *  ejecutamos el paquete 
 *      npm run test
 * 
 *  Para que este al pendiente de los cambios en el archivo de pruebas
 *      "test": "jest --watchAll"
 * 
 *  Instalamos paquete para intelly Sense
 *  npm install -D @types/jest
 * 
 *  Instalación de dependencias para compatibilidad con javaScript moderno
 *  npm install -D babel-jest @babel/core @babel/preset-env
 */