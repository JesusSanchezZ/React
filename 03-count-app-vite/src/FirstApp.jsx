// import { Fragment } from 'react';

// export const FirstApp = () => {
//     return (
//         <Fragment>
//             <h1>First App</h1>
//             <p>Hola a todos</p>
//         </Fragment>
//     );
// }
import PropTypes from 'prop-types';

const newMessage = {
    nombre: 'Jesus',
    mensaje: 'Hola a todos'
};

const saludo = () =>{
    const nombre = newMessage.nombre;
    const mensaje = newMessage.mensaje;

    return(
        <>
            <h1>{nombre} <span>Les dice:</span></h1>
            <p><strong>{mensaje}</strong></p>
        </>
    );
}

const getResult = (a, b) => {
    return a + b;
}

export const FirstApp = ( { title, subTitle, name } ) => {
    //console.log( props );
    
    return(
        <>
            <h1 data-testid="test-title"> { title } </h1>
            {/* <h1>{ JSON.stringify(newMessage) }</h1> */}
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    //title: 'No hay título',
    subTitle: 'No hay subtítulo',
    name: 'Jesus Sanchez'
}

/**
 * instalar react testing library
 * npm install -D @testing-library/react
 */