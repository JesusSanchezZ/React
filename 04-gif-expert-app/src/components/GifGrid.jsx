// import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from "prop-types";


export const GrifGrid = ({category}) =>{
    const { images, isLoading } = useFetchGifs( category );

    //console.log({isLoading});
    // const [images, setImages] = useState([]);
    // const getImages = async () =>{
    //     const imgs = await getGifs(category);
    //     setImages(imgs);
    // }

    // useEffect( () =>{
    //     getImages();
    // },[] )

    return(
        <>
            <h3>{ category }</h3>

            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className="card-grid">
                {
                images.map( (image) => (
                    <GifItem
                        key={image.id}
                        // title={image.title}
                        // url={image.url}
                        {...image}
                    />
                ))
                }
            </div>
        </>
    )
}

GrifGrid.propTypes = {
    category: PropTypes.string.isRequired
}