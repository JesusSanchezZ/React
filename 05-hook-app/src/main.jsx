import React from 'react';
import ReactDOM from 'react-dom/client';
import { BrowserRouter } from 'react-router-dom';

// import { Padre } from './07-tarea-memo/Padre';
import { MainApp } from './09-useContext/MainApp';
//import { CallbackHook } from './06-memos/CallbackHook';
//import { MemoHook } from './06-memos/MemorizeHook';
//import { Memorize } from './06-memos/Memorize';
//import { Layout } from './05-useLayoutEffect/Layout';
//import { FocusScreen } from './04-useRef/FocusScreen';
//import { HooksApp } from './HooksApp'
//import { CounterApp } from './01-useState/CounterApp';
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
//import { SimpleForm } from './02-useEffect/SimpleForm';
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
//import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
import './index.css';

//import './08-useReducer/intro-reducer';           // presentacion de useReducer
//import { TodoApp } from './08-useReducer/TodoApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    {/* <React.StrictMode> */}
    <MainApp />
    {/* </React.StrictMode> */}
  </BrowserRouter>
)

// instalacion de Router app
// npm install react-router-dom@6
