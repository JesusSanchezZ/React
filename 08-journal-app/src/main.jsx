import React from 'react';
import ReactDOM from 'react-dom/client';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { JournalApp } from './JournalApp';
import { store } from './store';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Provider store={ store }>
      <BrowserRouter>
        <JournalApp />
      </BrowserRouter>
    </Provider>
  </React.StrictMode>
)



/**
 * Preparacíon del proyecto:
 * creación del proyecto: npm create vite
 *  cd "nombre del proyecto"
 *  npm install
 *  ejecución del proyecto: npm run dev  // modo desarrollo
 * 
 * Instalación de React Router
 *  npm install react-router-dom@6       // instala la versión 6 de React router
 * 
 * Instalación de material ui: https://mui.com
 *  npm install @mui/material @emotion/react @emotion/styled
 * Agregar al index.html la fuente Roboto
 *  <link
 *    rel="stylesheet"
 *    href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
 *  />
 *  
 * Instalación de iconos
 *  npm install @mui/icons-material
 * 
 * Instalación de Redux Toolkit
 *  npm install @reduxjs/toolkit react-redux
 * 
 * Instalación de 
 */
