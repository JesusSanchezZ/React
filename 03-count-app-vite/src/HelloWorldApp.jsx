// function App() {
//     return (<h1>¡¡Hola Mundo!!</h1>);
// }

// export default App;

import { FirstApp } from './FirstApp';

export const HelloWorldApp = () => {
    return (
    <div>
        <h1>Hola mundo</h1>
        <FirstApp />
    </div>);
}