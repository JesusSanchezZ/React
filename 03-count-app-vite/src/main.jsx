import React from "react";
import ReactDOM from "react-dom/client";
//import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';

import './styles.css';

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={ Math.round(Math.random() * 300 - 150) } />
        {/* <FirstApp title="Hola, soy yo"></FirstApp> */}
    </React.StrictMode>
);


/**
 * Creación de aplicación react con vite
 * npm create vite
 * --> npm install
 *      npm run dev
 */