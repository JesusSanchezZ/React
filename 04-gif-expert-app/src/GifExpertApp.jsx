import { useState } from "react";

import { AddCategory, GrifGrid } from './components';
// import { AddCategory } from './components/AddCategory';
// import { GrifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState([ 'One Punch']);
    //console.log(categories);
    const onAddCategory = ( newCategory ) => {
        if( categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories]);
        //console.log(newCategory);
        //categories.push(newCategory);
        // const entrada = document.querySelector('input');
        //console.log(entrada);
        // setCategories( cat => [...cat, 'Valor']);
        // setCategories([entrada.value,...categories]);
    }

    return(
        <>
            {/* TÃ­tulo */}
            <h1>GifExpertApp</h1>

            {/* Input */}
            <AddCategory //setCategories={ setCategories }
                onNewCategory={event => onAddCategory(event) }
            />

            {/* Listado de Gif */}
            {categories.map(category =>
            (
                <GrifGrid key={category} category={category} />
            )
            )}
        </>
    )
}