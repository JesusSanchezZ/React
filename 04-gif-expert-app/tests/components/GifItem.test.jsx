import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Pruebas a componente <GifItems/>',()=>{
    const title='Saytama';
    const url  ='https://udemy.com/saytama.gif';

    test('Snapshot inicial',()=>{
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el url y el alt indicado',()=>{
        render(<GifItem title={title} url={url}/>);
        //screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( alt ).toBe( title );
        expect( src ).toBe( url );
    });

    test('Debe de mostrar el tÃ­tulo en el componente', ()=>{
        render(<GifItem title={title} url={url}/>);
        expect( screen.getByText( title ) ).toBeTruthy();
    });
});