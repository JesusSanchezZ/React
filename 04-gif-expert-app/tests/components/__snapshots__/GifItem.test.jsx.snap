// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente <GifItems/> Snapshot inicial 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saytama"
      src="https://udemy.com/saytama.gif"
    />
    <p>
      Saytama
    </p>
  </div>
</div>
`;
